lib/elixir/src/elixir_errors.erl

```diff
@@ -12,6 +12,12 @@
 warn(none, File, Warning) ->
   warn(0, File, Warning);
 warn(Line, File, Warning) when is_integer(Line), is_binary(File) ->
+  CompilerPid = get(elixir_compiler_pid),
+  if
+    CompilerPid =/= undefined ->
+      CompilerPid ! {warning, File, Line, list_to_binary(Warning)};
```

Comment:
Maybe we should keep it as char data and not do any conversion since we end-up calling `to_charlist` later on anyway?